### PR TITLE
test/robustness: Create dedicated traffic package

### PR DESCRIPTION
**Pull request by serathius**
cc @ahrtr @ptabor

### CODE REVIEW

**Comment by lavacat** [comment]
Seams like there are couple minor refactors besides creating traffic package:
- `config FailpointConfig` was removed, I assume for better readability
- `type watchConfig` was added, not clear why

Can you add brief list of minor refactors in commit body?

In terms of traffic package - I didn't review actual logic, assuming this is copy/paste. 
Overall LGTM

File: tests/robustness/linearizability_test.go [hunk 1]
```diff
@@ -149,50 +69,53 @@ func TestRobustness(t *testing.T) {
...
 	})
 	scenarios = append(scenarios, scenario{
 		name:      "Issue15220",
 		failpoint: RandomFailpoint,
-		traffic:   &ReqProgTraffic,
-		config: *e2e.NewConfig(
+		watch: watchConfig{
+			requestProgress: true,
+		},
+		traffic: &traffic.HighTraffic,
```
Comment (ahrtr) [inline tests/robustness/linearizability_test.go:108]: Could you resolve all unrelated changes in a separate PR?  I expect this PR only focuses on moving all traffic related implementation into a dedicated package.

Comment (serathius) [reply tests/robustness/linearizability_test.go]: Done

**Comment by serathius** [comment]
Removed the additional changes. PTAL

**Review by ahrtr — APPROVED** [review]
LGTM

Thanks @serathius
